components/org.wso2.carbon.identity.api.server.api.resource/org.wso2.carbon.identity.api.server.api.resource.v1/src/main/java/org/wso2/carbon/identity/api/server/api/resource/v1/util/APIResourceMgtEndpointUtil.java

```diff
@@ -0,0 +1,178 @@
+ */
+public class APIResourceMgtEndpointUtil {
+
+    private static final Log log = LogFactory.getLog(APIResourceMgtEndpointUtil.class);
+
+    public static void validateAPIResource(APIResource apiResource) {
+
+        if (StringUtils.isBlank(apiResource.getName())) {
+            throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_API_RESOURCE_NAME);
+        }
+        if (StringUtils.isBlank(apiResource.getIdentifier())) {
+            throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_API_RESOURCE_IDENTIFIER);
+        }
+    }
+
+    public static void validateScopes(List<Scope> scopes) {
+
+        for (Scope scope : scopes) {
+            // Validate scope name.
+            if (StringUtils.isBlank(scope.getName())) {
+                throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_SCOPE_NAME);
+            }
+            // Validate restricted scope names.
+            List<String> restrictedScopes =
+                    IdentityUtil.getPropertyAsList(APIResourceMgtEndpointConstants.RESTRICTED_OAUTH2_SCOPES);
+            for (String restrictedScope : restrictedScopes) {
+                if (scope.getName().startsWith(restrictedScope)) {
+                    throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_RESTRICTED_SCOPE_NAME);
+                }
+            }
+
+            // Validate OIDC scopes.
+            try {
+                List<String> registeredOIDCScopes = APIResourceManagementServiceHolder.getOAuthAdminServiceImpl()
+                        .getRegisteredOIDCScope(ContextLoader.getTenantDomainFromContext());
+                if (registeredOIDCScopes.contains(scope.getName())) {
+                    throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_RESTRICTED_OIDC_SCOPES);
+                }
+            } catch (IdentityOAuthAdminException e) {
+                throw handleException(Response.Status.INTERNAL_SERVER_ERROR, ERROR_CODE_VALIDATE_SCOPES);
+            }
+        }
+    }
+
+    /**
+     * Validate the attributes provided for search.
+     *
+     * @param attributes List of attributes to be validated.
+     * @return List of validated attributes.
+     */
+    public static List<String> validateAndConvertToLowerCase(List<String> attributes) {
+        List<String> validatedAttributes = new ArrayList<>();
+
+        if (attributes != null && !attributes.isEmpty()) {
+            for (String attribute : attributes) {
+                String lowerCaseAttribute = attribute.toLowerCase(Locale.ENGLISH);
+                if (!APIResourceMgtEndpointConstants.ALLOWED_SEARCH_ATTRIBUTES.contains(lowerCaseAttribute)) {
+                    throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_SEARCH_ATTRIBUTE);
+                } else {
+                    validatedAttributes.add(lowerCaseAttribute);
+                }
+            }
+        }
+        return validatedAttributes;
```

Comment:
Organize the method logic so that it will not grow to the right.

```suggestion
       if (attributes == null) {
           return Collections.emptyList();
       }

        List<String> validatedAttributes = new ArrayList<>();
        for (String attribute : attributes) {
            String lowerCaseAttribute = attribute.toLowerCase(Locale.ENGLISH);
            if (!APIResourceMgtEndpointConstants.ALLOWED_SEARCH_ATTRIBUTES.contains(lowerCaseAttribute)) {
                    throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_SEARCH_ATTRIBUTE);
            } 
            validatedAttributes.add(lowerCaseAttribute);        
        }
        return validatedAttributes;
```

Reply:
Fixed with https://github.com/wso2/identity-api-server/pull/473/commits/5c51c9eff45d1c55a16c7858a056b5a9eed2ce86